src/App.tsx

```diff
@@ -1,14 +1,79 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { debounce } from 'lodash';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { Loader } from './components/Loader';
+import { FilterType } from './types/FilterType';
+
+const getVisibleTodos = (todos: Todo[], type: string, query: string) => {
+  let preparedTodos = [...todos];
+
+  if (query) {
+    const lowerQuery = query.toLowerCase();
+
+    preparedTodos = preparedTodos
+      .filter(todo => todo.title.toLowerCase().includes(lowerQuery));
+  }
+
+  preparedTodos = preparedTodos.filter(todo => {
+    switch (type) {
+      case FilterType.All:
+        return true;
+      case FilterType.Active:
+        return !todo.completed;
+      case FilterType.Completed:
+        return todo.completed;
+      default:
+        throw new Error('Filter type is incorrect');
+    }
+  });
+
+  return preparedTodos;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.All);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(true);
+  const [isLoaded, setIsLoaded] = useState(false);
```

Comment:
let us follow DRY isLoaded = !isLoading, try to use one state for handle loading
```suggestion
  const [isLoading, setIsLoading] = useState(true);
```

Reply:
Yep, changed)